### PR TITLE
test: add IotAuthClientFake

### PR DESCRIPTION
**Pull request by jbutler**
**Issue #, if available:**

**Description of changes:**
Adds IotAuthClientFake which can be used during testing to avoid mocking IotAuthClient. More work is needed to remove mocks from some of the IPC tests.

**Why is this change necessary:**

**How was this change tested:**

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/integrationtests/java/com/aws/greengrass/integrationtests/ipc/AuthorizeClientDeviceActionTest.java [hunk 1]
```diff
@@ -82,17 +77,17 @@ private static void authzClientDeviceAction(GreengrassCoreIPCClient ipcClient,
...
         // Set this property for kernel to scan its own classpath to find plugins
         System.setProperty("aws.greengrass.scanSelfClasspath", "true");
         kernel = new Kernel();
-        kernel.getContext().put(GreengrassServiceClientFactory.class, clientFactory);
-
-        when(clientFactory.fetchGreengrassV2DataClient()).thenReturn(v2DataClient);
 
+        // Set up Iot auth client
+        IotAuthClientFake iotAuthClientFake = new IotAuthClientFake();
+        kernel.getContext().put(IotAuthClient.class, iotAuthClientFake);
```
Comment (jbutler) [inline src/integrationtests/java/com/aws/greengrass/integrationtests/ipc/AuthorizeClientDeviceActionTest.java:90]: TODO: these are not even close to being actual integration tests. We should remove the rest of these mocks, but will do that in a different PR (we have separate tasks for these integ tests)

**Review by MikeDombo — DISMISSED** [review]

File: src/test/java/com/aws/greengrass/clientdevices/auth/iot/IotAuthClientFake.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+        activeCertIds.remove(certId);
+        inactiveCertIds.add(certId);
+    }
+
+    public void attachCertificateToThing(String thing, String certPem) throws InvalidCertificateException {
+        Certificate cert = Certificate.fromPem(certPem);
+        String certId = cert.getCertificateId();
+
+        if (!thingToCerts.containsKey(thing)) {
```
Comment (jcosentino11) [inline src/test/java/com/aws/greengrass/clientdevices/auth/iot/IotAuthClientFake.java:43]:
```suggestion
        thingToCerts.computeIfAbsent(thing, k -> new HashSet<>())
                .add(certId);
```

**Review by jcosentino11 — APPROVED** [review]
